adoc/deployment-azure.adoc

```diff
@@ -0,0 +1,204 @@
+  * By design it's not possible to leverage the bastion host without using the
+    ssh session embedded into the browser. This makes impossible to use tools like
+    `sftp` or `scp`.
+  * You have to rely on copy and paste to share data (like the `admin.conf` file
+    generated by skuba) between the remote nodes and your local system.
+    You can "rely" on `cat`, `base64` and a lot of copy and paste...
+  * `skuba` requires a private ssh key to connect to all the nodes of the cluster.
+    You have to upload the private key you specified at cluster creation
+    (ugh!) or create a new one inside of the first master node and copy that
+    around the cluster.
+
+=== Deploying the Infrastructure
+
+On the management machine, find the {tf} template files for AZURE in
+`/usr/share/caasp/terraform/azure`. These files have been installed as part of
+the management pattern (`sudo zypper in -t pattern SUSE-CaaSP-Management`).
+
+Copy this folder to a location of your choice as the files need adjustment.
+
+----
+mkdir -p ~/caasp/deployment/
+cp -r /usr/share/caasp/terraform/azure/ ~/caasp/deployment/
+cd ~/caasp/deployment/azure/
+----
+
+Once the files are copied, rename the `terraform.tfvars.example` file to
+`terraform.tfvars`:
+
+----
+cp terraform.tfvars.example terraform.tfvars
+----
+
+Edit the `terraform.tfvars` file and add/modify the following variables:
+
+include::deployment-terraform-example.adoc[tags=tf_azure]
+
+[TIP]
+====
+You can set timezone and other parameters before deploying the nodes
+by modifying the cloud-init template:
+
+* `~/caasp/deployment/azure/cloud-init/cloud-init.yaml.tpl`
+====
+
+You can enter the registration code for your nodes in
+`~/caasp/deployment/azure/registration.auto.tfvars` instead of the
+`terraform.tfvars` file.
+
+Substitute `CAASP_REGISTRATION_CODE` for the code from <<registration_code>>.
+
+[source,json]
+----
+# SUSE CaaSP Product Key
+caasp_registry_code = "<CAASP_REGISTRATION_CODE>"
+----
+
+This is required so all the deployed nodes can automatically register
+with {scc} and retrieve packages.
+
+Now you can deploy the nodes by running:
+
+----
+terraform init
+terraform plan
+terraform apply
+----
+
+Check the output for the actions to be taken. Type "yes" and confirm with
+kbd:[Enter] when ready.
+{tf} will now provision all the cluster infrastructure.
+
+.Note Down IP/FQDN For the Nodes
+[IMPORTANT]
+====
+The IP addresses and FQDN of the generated machines will be displayed in the
+{tf} output during the cluster node deployment. You need these information
+later to deploy {productname}.
+
+These information can be obtained at any time by executing the
+`terraform output` command within the directory from which you executed
+{tf}.
+====
+
+=== Logging into the Cluster Nodes
+
+Connecting to the cluster nodes can be accomplished only via SSH key-based
+authentication thanks to the ssh-public key injection done earlier via
+`cloud-init`. You can use the predefined `sles` user to log in.
```

Comment:
It's technically not using cloud-init yet, but it will as soon as we rebase on top of SLE 15 SP2. I wouldn't bother to change that right now

Reply:
changed